huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go

```diff
@@ -478,6 +560,41 @@ func updateServiceEnterpriseId(ctx context.Context, client *golangsdk.ServiceCli
 	return doingUpdate(ctx, client, opts, d.Timeout(schema.TimeoutUpdate))
 }
 
+func buildAuthConfig(configInfo map[string]interface{}) *services.OtpConfigInfo {
+	result := services.OtpConfigInfo{
+		Enable:               utils.Bool(configInfo["enable"].(bool)),
+		ReceiveMode:          configInfo["receive_mode"].(string),
+		AuthUrl:              configInfo["auth_url"].(string),
+		AppId:                configInfo["app_id"].(string),
+		AppSecrte:            configInfo["app_secret"].(string),
+		AuthServerAccessMode: configInfo["auth_server_access_mode"].(string),
+		CertContent:          configInfo["cert_content"].(string),
+		ApplyRule: &services.ApplyRule{
+			RuleType: configInfo["rule_type"].(string),
+			Rule:     configInfo["rule"].(string),
+		},
+	}
+
+	return &result
+}
+
+func updateAssistAuthConfig(client *golangsdk.ServiceClient, d *schema.ResourceData) error {
+	authConfig := d.Get("otp_config_info").([]interface{})
+	if len(authConfig) < 1 {
+		return nil
+	}
+
+	opts := services.UpdateAuthConfigOpts{
+		AuthType:      "OTP",
+		OptConfigInfo: buildAuthConfig(authConfig[0].(map[string]interface{})),
+	}
+	err := services.UpdateAssistAuthConfig(client, opts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
```suggestion
func updateAssistAuthConfig(client *golangsdk.ServiceClient, d *schema.ResourceData) error {
	authConfig := d.Get("otp_config_info").([]interface{})
	if len(authConfig) < 1 {
		return nil
	}

	opts := services.UpdateAuthConfigOpts{
		AuthType:      "OTP",
		OptConfigInfo: buildAuthConfig(authConfig[0].(map[string]interface{})),
	}
	return services.UpdateAssistAuthConfig(client, opts)
}
```
Both function updateAssistAuthConfig and function UpdateAssistAuthConfig are return the result of error type, there is unnecessary to check whether function error response is not nil.

Reply:
fixed